### PR TITLE
Take all the snapshots into account in the style system

### PR DESCRIPTION
**Pull request by emilio**
See [bug 1355343](https://bugzilla.mozilla.org/show_bug.cgi?id=1355343).

The servo part of this patch presumably needs some polishing, let's see.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16778)

### CODE REVIEW

**Comment by emilio** [comment]
@bors-servo retry

 * Lockfile.

**Comment by emilio** [comment]
@bors-servo try

 *

**Comment by emilio** [comment]
@bors-servo try

 * Tidy....

**Comment by emilio** [comment]
@bors-servo retry

 * `LLVM ERROR: Invalid data was encountered while parsing the file`. wat?

**Comment by jdm** [comment]
```
  ▶ TIMEOUT [expected OK] /_mozilla/css/animations/transition-raf.html
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  └ 3.3 (Core Profile) Mesa 12.0.1

  ▶ Unexpected subtest result in /_mozilla/css/animations/transition-raf.html:
  │ TIMEOUT [expected PASS] Transitions should work during RAF loop
  └   → Test timed out

  ▶ Unexpected subtest result in /_mozilla/mozilla/out-of-order-stylesheet-loads-and-imports.html:
  │ FAIL [expected PASS] out-of-order stylesheet loads for the same element happen correctly, even with imports
  │   → assert_equals: expected "rgb(0, 128, 0)" but got "rgba(0, 0, 0, 0)"
  │ 
  │ @http://web-platform.test:8000/_mozilla/mozilla/out-of-order-stylesheet-loads-and-imports.html:14:5
  │ Test.prototype.step_timeout/<@http://web-platform.test:8000/resources/testharness.js:1464:20
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1406:20
  └ Test.prototype.step_func/<@http://web-platform.test:8000/resources/testharness.js:1430:20

  ▶ Unexpected subtest result in /_mozilla/mozilla/out-of-order-stylesheet-loads.html:
  │ FAIL [expected PASS] out-of-order stylesheet loads for the same element happen correctly
  │   → assert_equals: expected "rgb(0, 128, 0)" but got "rgba(0, 0, 0, 0)"
  │ 
  │ @http://web-platform.test:8000/_mozilla/mozilla/out-of-order-stylesheet-loads.html:14:5
  │ Test.prototype.step_timeout/<@http://web-platform.test:8000/resources/testharness.js:1464:20
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1406:20
  └ Test.prototype.step_func/<@http://web-platform.test:8000/resources/testharness.js:1430:20

  ▶ FAIL [expected PASS] /html/semantics/document-metadata/the-link-element/stylesheet-change-href.html
  └   → /html/semantics/document-metadata/the-link-element/stylesheet-change-href.html e156db2a536952daa81c819cc5670f1554fa319c
/html/semantics/document-metadata/the-link-element/stylesheet-change-href-ref.html d4fd1d548349918fbb8d5497c9ce4a7007346b6f
Testing e156db2a536952daa81c819cc5670f1554fa319c == d4fd1d548349918fbb8d5497c9ce4a7007346b6f

  ▶ FAIL [expected PASS] /html/semantics/document-metadata/the-link-element/stylesheet-with-base.html
  └   → /html/semantics/document-metadata/the-link-element/stylesheet-with-base.html 23b755622cc81841a23cc8431a6546dfb1401c66
/html/semantics/document-metadata/the-link-element/stylesheet-with-base-ref.html 16698156583755674f3837b45ba940d68a1d3b1f
Testing 23b755622cc81841a23cc8431a6546dfb1401c66 == 16698156583755674f3837b45ba940d68a1d3b1f

  ▶ FAIL [expected PASS] /_mozilla/css/background_external_stylesheet.html
  └   → /_mozilla/css/background_external_stylesheet.html 23b755622cc81841a23cc8431a6546dfb1401c66
/_mozilla/css/background_rust_ref.html 93a863736bf0274a0616cceaddca8071a847c80e
Testing 23b755622cc81841a23cc8431a6546dfb1401c66 == 93a863736bf0274a0616cceaddca8071a847c80e

  ▶ FAIL [expected PASS] /_mozilla/css/iframe/hide_and_show.html
  └   → /_mozilla/css/iframe/hide_and_show.html 23b755622cc81841a23cc8431a6546dfb1401c66
/_mozilla/css/iframe/hide_and_show_ref.html df80fe8f89a9c48e16251483ba2181b39b82df44
Testing 23b755622cc81841a23cc8431a6546dfb1401c66 == df80fe8f89a9c48e16251483ba2181b39b82df44

  ▶ FAIL [expected PASS] /_mozilla/css/height_compute_reset.html
  └   → /_mozilla/css/height_compute_reset.html 12eaeb3754f718059b52aeff039a4011f2987167
/_mozilla/css/height_compute.html 9159cbbcde5eb29ea3765d6fb6d83f584331ae96
Testing 12eaeb3754f718059b52aeff039a4011f2987167 == 9159cbbcde5eb29ea3765d6fb6d83f584331ae96

  ▶ FAIL [expected PASS] /_mozilla/css/incremental_visibility_a.html
  └   → /_mozilla/css/incremental_visibility_a.html 23b755622cc81841a23cc8431a6546dfb1401c66
/_mozilla/css/incremental_visibility_ref.html 7949e80732fa0f4afdbae3b1cad96ccd75bb04f6
Testing 23b755622cc81841a23cc8431a6546dfb1401c66 == 7949e80732fa0f4afdbae3b1cad96ccd75bb04f6

  ▶ FAIL [expected PASS] /_mozilla/css/incremental_text_color_a.html
  └   → /_mozilla/css/incremental_text_color_a.html 71e7a1e4b5c5781089d3aa5e996410103431401a
/_mozilla/css/incremental_text_color_ref.html 1749d8269fc416224d0c90117d9fa3a982359690
Testing 71e7a1e4b5c5781089d3aa5e996410103431401a == 1749d8269fc416224d0c90117d9fa3a982359690

  ▶ FAIL [expected PASS] /_mozilla/css/inline_block_parent_padding_a.html
  └   → /_mozilla/css/inline_block_parent_padding_a.html 8c9c89a6a5db8edefb7814fa5f83b44dddc7f39b
/_mozilla/css/inline_block_parent_padding_ref.html 495641fc705b6af91aa9ff0d3235cd2c5342179c
Testing 8c9c89a6a5db8edefb7814fa5f83b44dddc7f39b == 495641fc705b6af91aa9ff0d3235cd2c5342179c

  ▶ FAIL [expected PASS] /_mozilla/css/inline_margins_a.html
  └   → /_mozilla/css/inline_margins_a.html 94233d3083f8324c3fd3598a971eec566e479d3e
/_mozilla/css/inline_margins_ref.html 98b9582cd0b6783d096b7b75adca05d08c875c14
Testing 94233d3083f8324c3fd3598a971eec566e479d3e == 98b9582cd0b6783d096b7b75adca05d08c875c14

  ▶ FAIL [expected PASS] /_mozilla/css/letter_spacing_a.html
  └   → /_mozilla/css/letter_spacing_a.html 3f35a6ecb734ec70935867893e9aaf101e54ab0d
/_mozilla/css/letter_spacing_ref.html ab12457fb941eb8f6b65fbdd75de9a9fde8d97ac
Testing 3f35a6ecb734ec70935867893e9aaf101e54ab0d == ab12457fb941eb8f6b65fbdd75de9a9fde8d97ac

  ▶ FAIL [expected PASS] /_mozilla/css/negation-attr-dependence.html
  └   → /_mozilla/css/negation-attr-dependence.html 94004ce3339acee5311254cef2fc9eb9d1cd225d
/_mozilla/css/negation-attr-dependence-ref.html c29e6c373ffe302706a87625fd354b3dbe1d4cb9
Testing 94004ce3339acee5311254cef2fc9eb9d1cd225d == c29e6c373ffe302706a87625fd354b3dbe1d4cb9

  ▶ FAIL [expected PASS] /_mozilla/css/overflow_wrap_a.html
  └   → /_mozilla/css/overflow_wrap_a.html fa0b18771852fda535dd4f3a92a41d5042bf5f20
/_mozilla/css/overflow_wrap_ref.html aa59b6aaebca0803219975377c889eb5833a4e14
Testing fa0b18771852fda535dd4f3a92a41d5042bf5f20 == aa59b6aaebca0803219975377c889eb5833a4e14

  ▶ FAIL [expected PASS] /_mozilla/css/restyle_hints_attr.html
  └   → /_mozilla/css/restyle_hints_attr.html 26aec255ebd9f0f6bf980024b006e2a8cafb3c8d
/_mozilla/css/restyle_hints_attr_ref.html 951f272e45cd9df72c92a8389aec19ba20d5fb1c
Testing 26aec255ebd9f0f6bf980024b006e2a8cafb3c8d == 951f272e45cd9df72c92a8389aec19ba20d5fb1c

  ▶ FAIL [expected PASS] /_mozilla/css/rotate_anim.html
  └   → /_mozilla/css/rotate_anim.html cbd43d592e3765186960f4451430087a024b1946
/_mozilla/css/rotate_anim_ref.html f16bb06405c422ebffa878e7b8144f439f28b71b
Testing cbd43d592e3765186960f4451430087a024b1946 == f16bb06405c422ebffa878e7b8144f439f28b71b

  ▶ FAIL [expected PASS] /_mozilla/css/restyle_hints_state.html
  └   → /_mozilla/css/restyle_hints_state.html 184baddc69073c1dd7ddd75a070e7383d2b36801
/_mozilla/css/restyle_hints_state_ref.html 051dc10c1a628262ae5d28ec7a45d2b75ddcf195
Testing 184baddc69073c1dd7ddd75a070e7383d2b36801 == 051dc10c1a628262ae5d28ec7a45d2b75ddcf195

  ▶ FAIL [expected PASS] /_mozilla/css/table_specified_width_a.html
  └   → /_mozilla/css/table_specified_width_a.html e4f0da49481fe3230998603ed4b50f56f66b37fd
/_mozilla/css/table_specified_width_ref.html f4b347ca6ae0b047dbbeebd6297d6412e2b910f9
Testing e4f0da49481fe3230998603ed4b50f56f66b37fd == f4b347ca6ae0b047dbbeebd6297d6412e2b910f9

  ▶ FAIL [expected PASS] /_mozilla/css/text_align_justify_a.html
  └   → /_mozilla/css/text_align_justify_a.html 55ad521c2c5c548e713497ce6555dd0e069db9a4
/_mozilla/css/text_align_justify_ref.html 05b606fff7f594f31359de08506dc85b4d42f4e7
Testing 55ad521c2c5c548e713497ce6555dd0e069db9a4 == 05b606fff7f594f31359de08506dc85b4d42f4e7

  ▶ FAIL [expected PASS] /_mozilla/css/text_indent_a.html
  └   → /_mozilla/css/text_indent_a.html affebb450392f77393d6c550290f4fb6a6dad8f1
/_mozilla/css/text_indent_ref.html 764bafbb1fab27f34c79ae08dd9c0fe42ec80e7b
Testing affebb450392f77393d6c550290f4fb6a6dad8f1 == 764bafbb1fab27f34c79ae08dd9c0fe42ec80e7b

  ▶ FAIL [expected PASS] /_mozilla/css/text_shadow_simple_a.html
  └   → /_mozilla/css/text_shadow_simple_a.html a9953665181a32844a6671e0181bfef5dbc53918
/_mozilla/css/text_shadow_simple_ref.html 98d24de16515d5b688b711f41cfd050725c684a5
Testing a9953665181a32844a6671e0181bfef5dbc53918 == 98d24de16515d5b688b711f41cfd050725c684a5

  ▶ FAIL [expected PASS] /_mozilla/css/text_node_opacity.html
  └   → /_mozilla/css/text_node_opacity.html a38ade66f13b6176dfe0da956eaa0dfc549fcf00
/_mozilla/css/text_node_opacity_ref.html 3327305df51d54ccc8f570b996c5c63460b9aa1a
Testing a38ade66f13b6176dfe0da956eaa0dfc549fcf00 == 3327305df51d54ccc8f570b996c5c63460b9aa1a

  ▶ FAIL [expected PASS] /_mozilla/css/transition_calc.html
  └   → /_mozilla/css/transition_calc.html 80c420dd3659d2f3ddfe047fef0539039268e825
/_mozilla/css/transition_calc_ref.html 0643e7cd25c9183d2270bd63bd204e74920beda0
Testing 80c420dd3659d2f3ddfe047fef0539039268e825 == 0643e7cd25c9183d2270bd63bd204e74920beda0

  ▶ FAIL [expected PASS] /_mozilla/css/vertical_align_bottom_a.html
  └   → /_mozilla/css/vertical_align_bottom_a.html 4afd185805577b29911b8da23492555f2588e4ac
/_mozilla/css/vertical_align_bottom_ref.html ca5768b40bb98e80245c645058f40e5081230261
Testing 4afd185805577b29911b8da23492555f2588e4ac == ca5768b40bb98e80245c645058f40e5081230261

  ▶ FAIL [expected PASS] /_mozilla/css/vertical_align_super_a.html
  └   → /_mozilla/css/vertical_align_super_a.html d6c9e7db63ee2219a56a5c59528ca1a384807cc0
/_mozilla/css/vertical_align_super_ref.html 28cfc87e52778d78f19cff503de8c143a302df58
Testing d6c9e7db63ee2219a56a5c59528ca1a384807cc0 == 28cfc87e52778d78f19cff503de8c143a302df58

  ▶ FAIL [expected PASS] /_mozilla/css/vertical_align_text_bottom_a.html
  └   → /_mozilla/css/vertical_align_text_bottom_a.html e6663119ff6284446c6efd2506349a895a4f7fdb
/_mozilla/css/vertical_align_text_bottom_ref.html 9f76b536aa7963983ec96d572308591639546af0
Testing e6663119ff6284446c6efd2506349a895a4f7fdb == 9f76b536aa7963983ec96d572308591639546af0

  ▶ FAIL [expected PASS] /_mozilla/css/vertical_align_sub_a.html
  └   → /_mozilla/css/vertical_align_sub_a.html 1311b914b3ff497830eb7b91c7f8e4ddb5eec8a7
/_mozilla/css/vertical_align_sub_ref.html 1818669ef3e9c01f03d4d385990c1fa581903836
Testing 1311b914b3ff497830eb7b91c7f8e4ddb5eec8a7 == 1818669ef3e9c01f03d4d385990c1fa581903836

  ▶ FAIL [expected PASS] /_mozilla/css/vertical_align_text_top_a.html
  └   → /_mozilla/css/vertical_align_text_top_a.html 5207a1e0718ecbb0f118f0d0090666d893181288
/_mozilla/css/vertical_align_text_top_ref.html c2443aaa9eb0b82f57ff607fd4c464c75924458a
Testing 5207a1e0718ecbb0f118f0d0090666d893181288 == c2443aaa9eb0b82f57ff607fd4c464c75924458a

  ▶ FAIL [expected PASS] /_mozilla/css/vertical_align_top_a.html
  └   → /_mozilla/css/vertical_align_top_a.html c05fbd6c942c3fbdf5ae83ad4dbe551b959d3569
/_mozilla/css/vertical_align_top_ref.html b211a1afb063d9cbb74da396db6fd55185e8256d
Testing c05fbd6c942c3fbdf5ae83ad4dbe551b959d3569 == b211a1afb063d9cbb74da396db6fd55185e8256d

  ▶ FAIL [expected PASS] /_mozilla/css/vertical_align_top_span_a.html
  └   → /_mozilla/css/vertical_align_top_span_a.html e8d8f38b7dc81ec0e2ce7001c4964e1cb6918de7
/_mozilla/css/vertical_align_top_span_ref.html b3ee41a96edfd3df879b276c9af7add6afb251eb
Testing e8d8f38b7dc81ec0e2ce7001c4964e1cb6918de7 == b3ee41a96edfd3df879b276c9af7add6afb251eb

  ▶ FAIL [expected PASS] /_mozilla/css/vertical_align_top_bottom_a.html
  └   → /_mozilla/css/vertical_align_top_bottom_a.html 026654b82abc983e288f0311e02b4debf5b32084
/_mozilla/css/vertical_align_top_bottom_ref.html 6b9716c90a5059fc250ce0b687416eccec53911b
Testing 026654b82abc983e288f0311e02b4debf5b32084 == 6b9716c90a5059fc250ce0b687416eccec53911b

  ▶ FAIL [expected PASS] /_mozilla/css/word_break_a.html
  └   → /_mozilla/css/word_break_a.html 7aba30ba46e9390a45dc380411c106742c64df01
/_mozilla/css/word_break_ref.html 9c715deb6ed592dfe5cda0c52afa2ff6d9f1884c
Testing 7aba30ba46e9390a45dc380411c106742c64df01 == 9c715deb6ed592dfe5cda0c52afa2ff6d9f1884c

  ▶ FAIL [expected PASS] /_mozilla/css/word-spacing.html
  └   → /_mozilla/css/word-spacing.html d8c6766ef2007b455d1c61a1ed284f3e21aaa3d0
/_mozilla/css/word-spacing-ref.html a1870b0df7e4e3e3c82f38092a339c469a39df8e
Testing d8c6766ef2007b455d1c61a1ed284f3e21aaa3d0 == a1870b0df7e4e3e3c82f38092a339c469a39df8e

  ▶ FAIL [expected PASS] /_mozilla/mozilla/simple_scroll_to_fragment.html
  └   → /_mozilla/mozilla/simple_scroll_to_fragment.html 2d5103c1d79069b6628ffe931d05456e5b9244b2
/_mozilla/mozilla/simple_scroll_to_fragment_ref.html 7949e80732fa0f4afdbae3b1cad96ccd75bb04f6
Testing 2d5103c1d79069b6628ffe931d05456e5b9244b2 == 7949e80732fa0f4afdbae3b1cad96ccd75bb04f6

  ▶ FAIL [expected PASS] /_mozilla/mozilla/stylesheet-adopt-panic.html
  └   → /_mozilla/mozilla/stylesheet-adopt-panic.html aa7c29113837a443cdc2f4e97b98ab7d18b45f6d
/_mozilla/mozilla/stylesheet-adopt-panic-ref.html 015a11ff3ffbfe99fe5e69a815b2a7600caf86e1
Testing aa7c29113837a443cdc2f4e97b98ab7d18b45f6d == 015a11ff3ffbfe99fe5e69a815b2a7600caf86e1

  ▶ Unexpected subtest result in /html/dom/elements/global-attributes/id-attribute.html:
  │ FAIL [expected PASS] After setting id via setAttribute attribute, CSS association is via the new ID.
  │   → assert_equals: expected "5" but got "1"
  │ 
  │ @http://web-platform.test:8000/html/dom/elements/global-attributes/id-attribute.html:125:9
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1406:20
  │ test@http://web-platform.test:8000/resources/testharness.js:497:9
  └ @http://web-platform.test:8000/html/dom/elements/global-attributes/id-attribute.html:124:5

  ▶ Unexpected subtest result in /html/dom/elements/global-attributes/id-attribute.html:
  │ FAIL [expected PASS] After setting id via id attribute, CSS association is via the new ID.
  │   → assert_equals: expected "4" but got "1"
  │ 
  │ @http://web-platform.test:8000/html/dom/elements/global-attributes/id-attribute.html:108:9
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1406:20
  │ test@http://web-platform.test:8000/resources/testharness.js:497:9
  └ @http://web-platform.test:8000/html/dom/elements/global-attributes/id-attribute.html:107:5

  ▶ Unexpected subtest result in /subresource-integrity/subresource-integrity.sub.html:
  │ FAIL [expected PASS] Style: Same-origin with correct sha256 hash
  │   → assert_equals: expected "rgb(255, 255, 0)" but got "rgba(0, 0, 0, 0)"
  │ FAIL [expected PASS] Style: Same-origin with correct sha384 hash
  │   → assert_equals: expected "rgb(255, 255, 0)" but got "rgba(0, 0, 0, 0)"
  │ FAIL [expected PASS] Style: Same-origin with correct sha512 hash
  │   → assert_equals: expected "rgb(255, 255, 0)" but got "rgba(0, 0, 0, 0)"
  │ FAIL [expected PASS] Style: Same-origin with empty integrity
  │   → assert_equals: expected "rgb(255, 255, 0)" but got "rgba(0, 0, 0, 0)"
  │ FAIL [expected PASS] Style: Same-origin with multiple sha256 hashes, including correct.
  │   → assert_equals: expected "rgb(255, 255, 0)" but got "rgba(0, 0, 0, 0)"
  │ FAIL [expected PASS] Style: Same-origin with multiple sha256 hashes, including unknown algorithm.
  │   → assert_equals: expected "rgb(255, 255, 0)" but got "rgba(0, 0, 0, 0)"
  │ FAIL [expected PASS] Style: Same-origin with sha256 mismatch, sha512 match
  │   → assert_equals: expected "rgb(255, 255, 0)" but got "rgba(0, 0, 0, 0)"
  │ FAIL [expected PASS] Style: <crossorigin='anonymous'> with correct hash, ACAO: *
  │   → assert_equals: expected "rgb(255, 255, 0)" but got "rgba(0, 0, 0, 0)"
  │ FAIL [expected PASS] Style: <crossorigin='use-credentials'> with correct hash, CORS-eligible
  │   → assert_equals: expected "rgb(255, 255, 0)" but got "rgba(0, 0, 0, 0)"
  │ FAIL [expected PASS] Style: Cross-origin, empty integrity
  │   → assert_equals: expected "rgb(255, 255, 0)" but got "rgba(0, 0, 0, 0)"
  │ FAIL [expected PASS] Style: Same-origin with correct hash, options.
  │   → assert_equals: expected "rgb(255, 255, 0)" but got "rgba(0, 0, 0, 0)"
  │ FAIL [expected PASS] Style: Same-origin with unknown algorithm only.
  │   → assert_equals: expected "rgb(255, 255, 0)" but got "rgba(0, 0, 0, 0)"
  │ FAIL [expected PASS] Style: Same-origin with correct sha256 hash, rel='stylesheet license'
  │   → assert_equals: expected "rgb(255, 255, 0)" but got "rgba(0, 0, 0, 0)"
  │ FAIL [expected PASS] Style: Same-origin with correct sha256 hash, rel='license stylesheet'
  │   → assert_equals: expected "rgb(255, 255, 0)" but got "rgba(0, 0, 0, 0)"
  │ 
  │ SRIStyleTest.prototype.execute/</<@http://web-platform.test:8000/subresource-integrity/subresource-integrity.sub.html:136:21
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1406:20
  └ SRIStyleTest.prototype.execute/<@http://web-platform.test:8000/subresource-integrity/subresource-integrity.sub.html:134:17
```

File: components/script/dom/node.rs [hunk 1]
```diff
@@ -146,27 +146,36 @@ bitflags! {
...
 
         /// Whether any ancestor is a fragmentation container
         const CAN_BE_FRAGMENTED = 0x40,
         #[doc = "Specifies whether this node needs to be dirted when viewport size changed."]
-        const DIRTY_ON_VIEWPORT_SIZE_CHANGE = 0x80,
+        const DIRTY_ON_VIEWPORT_SIZE_CHANGE = 1 << 4,
 
         #[doc = "Specifies whether the parser has set an associated form owner for \
                  this element. Only applicable for form-associatable elements."]
-        const PARSER_ASSOCIATED_FORM_OWNER = 0x90,
```
Comment (emilio) [inline components/script/dom/node.rs]: Some funny thing, this flag was completely bogus before this patch. Wonder if it has any effect on the test results... Still investigating.

**Comment by jdm** [comment]
@bors-servo: retry
#16783

**Comment by jdm** [comment]
@bors-servo: retry
* silly cross3 builder

**Comment by emilio** [comment]
@bors-servo retry 

 * LLVM error again

**Comment by jdm** [comment]
```

  ▶ TIMEOUT [expected OK] /_mozilla/css/animations/transition-raf.html
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  └ 3.3 (Core Profile) Mesa 12.0.1

  ▶ Unexpected subtest result in /_mozilla/css/animations/transition-raf.html:
  │ TIMEOUT [expected PASS] Transitions should work during RAF loop
  └   → Test timed out


  ▶ Unexpected subtest result in /html/dom/elements/global-attributes/id-attribute.html:
  │ FAIL [expected PASS] After setting id via setAttribute attribute, CSS association is via the new ID.
  │   → assert_equals: expected "5" but got "1"
  │ 
  │ @http://web-platform.test:8000/html/dom/elements/global-attributes/id-attribute.html:125:9
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1406:20
  │ test@http://web-platform.test:8000/resources/testharness.js:497:9
  └ @http://web-platform.test:8000/html/dom/elements/global-attributes/id-attribute.html:124:5

  ▶ Unexpected subtest result in /html/dom/elements/global-attributes/id-attribute.html:
  │ FAIL [expected PASS] After setting id via id attribute, CSS association is via the new ID.
  │   → assert_equals: expected "4" but got "1"
  │ 
  │ @http://web-platform.test:8000/html/dom/elements/global-attributes/id-attribute.html:108:9
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1406:20
  │ test@http://web-platform.test:8000/resources/testharness.js:497:9
  └ @http://web-platform.test:8000/html/dom/elements/global-attributes/id-attribute.html:107:5

  ▶ FAIL [expected PASS] /_mozilla/css/iframe/hide_and_show.html
  └   → /_mozilla/css/iframe/hide_and_show.html 23b755622cc81841a23cc8431a6546dfb1401c66
/_mozilla/css/iframe/hide_and_show_ref.html df80fe8f89a9c48e16251483ba2181b39b82df44
Testing 23b755622cc81841a23cc8431a6546dfb1401c66 == df80fe8f89a9c48e16251483ba2181b39b82df44

  ▶ FAIL [expected PASS] /_mozilla/css/incremental_text_color_a.html
  └   → /_mozilla/css/incremental_text_color_a.html 71e7a1e4b5c5781089d3aa5e996410103431401a
/_mozilla/css/incremental_text_color_ref.html 1749d8269fc416224d0c90117d9fa3a982359690
Testing 71e7a1e4b5c5781089d3aa5e996410103431401a == 1749d8269fc416224d0c90117d9fa3a982359690

  ▶ FAIL [expected PASS] /_mozilla/css/incremental_visibility_a.html
  └   → /_mozilla/css/incremental_visibility_a.html 23b755622cc81841a23cc8431a6546dfb1401c66
/_mozilla/css/incremental_visibility_ref.html 7949e80732fa0f4afdbae3b1cad96ccd75bb04f6
Testing 23b755622cc81841a23cc8431a6546dfb1401c66 == 7949e80732fa0f4afdbae3b1cad96ccd75bb04f6

  ▶ FAIL [expected PASS] /_mozilla/css/negation-attr-dependence.html
  └   → /_mozilla/css/negation-attr-dependence.html 94004ce3339acee5311254cef2fc9eb9d1cd225d
/_mozilla/css/negation-attr-dependence-ref.html c29e6c373ffe302706a87625fd354b3dbe1d4cb9
Testing 94004ce3339acee5311254cef2fc9eb9d1cd225d == c29e6c373ffe302706a87625fd354b3dbe1d4cb9

  ▶ FAIL [expected PASS] /_mozilla/css/restyle_hints_attr.html
  └   → /_mozilla/css/restyle_hints_attr.html 26aec255ebd9f0f6bf980024b006e2a8cafb3c8d
/_mozilla/css/restyle_hints_attr_ref.html dd9372b41fc99dbbac902bab1dcaf28ed390544d
Testing 26aec255ebd9f0f6bf980024b006e2a8cafb3c8d == dd9372b41fc99dbbac902bab1dcaf28ed390544d

  ▶ FAIL [expected PASS] /_mozilla/css/rotate_anim.html
  └   → /_mozilla/css/rotate_anim.html cbd43d592e3765186960f4451430087a024b1946
/_mozilla/css/rotate_anim_ref.html f16bb06405c422ebffa878e7b8144f439f28b71b
Testing cbd43d592e3765186960f4451430087a024b1946 == f16bb06405c422ebffa878e7b8144f439f28b71b

  ▶ FAIL [expected PASS] /_mozilla/css/restyle_hints_state.html
  └   → /_mozilla/css/restyle_hints_state.html 184baddc69073c1dd7ddd75a070e7383d2b36801
/_mozilla/css/restyle_hints_state_ref.html d5ec4481f7293a7cfb4d44556f951af380694469
Testing 184baddc69073c1dd7ddd75a070e7383d2b36801 == d5ec4481f7293a7cfb4d44556f951af380694469

  ▶ FAIL [expected PASS] /_mozilla/css/transition_calc.html
  └   → /_mozilla/css/transition_calc.html 80c420dd3659d2f3ddfe047fef0539039268e825
/_mozilla/css/transition_calc_ref.html 0643e7cd25c9183d2270bd63bd204e74920beda0
Testing 80c420dd3659d2f3ddfe047fef0539039268e825 == 0643e7cd25c9183d2270bd63bd204e74920beda0

  ▶ FAIL [expected PASS] /_mozilla/mozilla/iframe/resize_after_load.html
  └   → /_mozilla/mozilla/iframe/resize_after_load.html 5424a0618386d87fdbe7dffb7c903d5b97eb9706
/_mozilla/mozilla/iframe/resize_after_load_ref.html 7949e80732fa0f4afdbae3b1cad96ccd75bb04f6
Testing 5424a0618386d87fdbe7dffb7c903d5b97eb9706 == 7949e80732fa0f4afdbae3b1cad96ccd75bb04f6
```

**Comment by jdm** [comment]
```
 ▶ CRASH [expected OK] /cssom-view/HTMLBody-ScrollArea_quirksmode.html
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  │ 3.3 (Core Profile) Mesa 12.0.1
  │ has_snapshot lied so badly (thread LayoutThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(1) }, at src/libcore/option.rs:794)
  │ 2017-05-09 22:13:35.392 servo[99272:1083243478] Metadata.framework [Error]: couldn't get the client port
  │ stack backtrace:
  │    0:        0x107663dde - backtrace::backtrace::trace::h1e1118131f7ae175
  │    1:        0x10766457c - backtrace::capture::Backtrace::new::h5f3de869c0349b3d
  │    2:        0x106a7a24d - servo::main::_$u7b$$u7b$closure$u7d$$u7d$::he1738f2aa8595d7e
  │    3:        0x1086aabf6 - std::panicking::rust_panic_with_hook::h2bd2f0b446f0face
  │    4:        0x1086aaa54 - std::panicking::begin_panic::h1732f61ecaaa434e
  │    5:        0x1086aa9c2 - std::panicking::begin_panic_fmt::h2948dd4668f99175
  │    6:        0x1086aa927 - rust_begin_unwind
  │    7:        0x1086d2ab0 - core::panicking::panic_fmt::hb2e726f3b579b19d
  │    8:        0x1086d2b1d - core::option::expect_failed::hf6ddaca0707dee4e
  │    9:        0x106bd9f82 - style::data::ElementData::compute_final_hint::hbc23f07d0f1e1a09
  │   10:        0x106b99389 - _$LT$layout..traversal..RecalcStyleAndConstructFlows$LT$$u27$a$GT$$u20$as$u20$style..traversal..DomTraversal$LT$E$GT$$GT$::process_preorder::h24e621e79f820ef6
  │   11:        0x106c24dfe - layout_thread::LayoutThread::handle_reflow::hcd4e25b9b75d44c4
  │   12:        0x106c1d299 - layout_thread::LayoutThread::handle_request_helper::h79825e530ec9b9b1
  │   13:        0x106c1aa24 - layout_thread::LayoutThread::handle_request::hf9228670e252110d
  │   14:        0x106bba27a - std::panicking::try::do_call::heb1467c028d9ef4d
  │   15:        0x1086abcca - __rust_maybe_catch_panic
  │   16:        0x106bd4974 - _$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$::call_box::h02afe2c3d71eca40
  │   17:        0x1086a79a5 - std::sys::imp::thread::Thread::new::thread_start::h5852519f9f48a1f2
  │   18:     0x7fff8655c059 - _pthread_body
  │   19:     0x7fff8655bfd6 - _pthread_start
  │ ERROR:servo: has_snapshot lied so badly
  │ Pipeline failed in hard-fail mode.  Crashing!
  └ thread panicked while processing panic. aborting.
```

**Comment by emilio** [comment]
@bors-servo r=bholley

 * Bogus flags were changing unexpected things under my feet.

**Comment by jdm** [comment]
@bors-servo: retry
* #16644

**Comment by bholley** [comment]
@bors-servo r-

We shouldn't land this until the VCS sync service is restored.

**Pull request merged**